woocommerce/google-listings-and-ads
# 1292: Fix fatal error when the plugin is activated before WooCommerce.

### Changes proposed in this Pull Request:

* Add requirement check in the plugins_loaded hook.
* Made ActivatedEvents implement Activateable interfaces. 
* Provide DELAYED_ACTIVATE option flag.
* Set DELAYED_ACTIVATE flag when PluginFactory::instance()->activate() has failed. 
* If DELAYED_ACTIVATE is set, run activate() on plugin_loaded and remove the DELAYED_ACTIVATE flag

Closes #1292.


### Screenshots:

WC not installed

![Screenshot 2022-03-17 at 16 31 07](https://user-images.githubusercontent.com/4209011/158828869-73673a7e-8c95-488e-ad99-40a5a4c57c2f.jpg)


### Detailed test instructions:


1. Installed both Google Listings & Ads and WooCommerce (both not activated yet).
2. Activate Google Listings & Ads first. You should see a notice stating that WC needs to be activated first.
3. Activate WooCommerce first, and then Google Listings & Ads. All good!

### Changelog entry

> Fix - Fatal error when the plugin is activated before WooCommerce.
